### PR TITLE
Add snapshot testing to contribution guide

### PR DESCRIPTION
**Pull request by j178**
## Summary

Add a `Snapshot testing` section to `CONTRIBUTING.md`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,6 +49,35 @@ cargo run python install
 
 The storage directory can be configured with `UV_PYTHON_INSTALL_DIR`.
 
+### Snapshot testing
+
+uv uses [insta](https://insta.rs/) for snapshot testing. It's recommended (but not necessary) to use
+`cargo-insta` for a better snapshot review experience:
```
Comment (charliermarsh) [inline CONTRIBUTING.md:55]: I would personally install this through `cargo install cargo-insta`. I'm curious to hear from others on the team.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -49,6 +49,35 @@ cargo run python install
...
+fn test_add() {
+    let context = TestContext::new("3.12");
+    uv_snapshot!(context.filters(), context.add().arg("pip"), @r"");
+}
+```
+
+To run and review a specific snapshot test:
+
+```shell
+cargo insta test --review --package <package> --test <test> -- <test_name> -- --exact
```
[the remark this answers is not in the document]
Comment (zanieb) [reply CONTRIBUTING.md:83]: I do use it with `--accept`, e.g.:

```
cit='cargo insta test --lib --bins --tests --accept --test-runner nextest -- '
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (zanieb) [reply CONTRIBUTING.md:55]: Yeah I would as well

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (charliermarsh) [reply CONTRIBUTING.md:83]: I'm mildly in favor of not suggesting it here (and just encouraging `cargo insta review`). I think it's a simpler mental model to view the `cargo insta` step as separate from testing itself.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -49,6 +49,33 @@ cargo run python install
...
+```
+
+To run and review a specific snapshot test:
+
+```shell
+cargo test --package <package> --test <test> -- <test_name> -- --exact
+cargo insta review
+
+# or combined into one command:
+cargo insta test --review --package <package> --test <test> -- <test_name> -- --exact
```
Comment (zanieb) [inline CONTRIBUTING.md:76]:
```suggestion
```

**Pull request merged**